src/main/java/ay2122s1_cs2103t_w16_2/btbb/logic/parser/client/FindClientCommandParser.java

```diff
@@ -55,21 +50,17 @@ public FindClientCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args,
                 PREFIX_CLIENT_NAME, PREFIX_CLIENT_PHONE, PREFIX_CLIENT_EMAIL, PREFIX_CLIENT_ADDRESS);
 
-        ClientPredicateCollection clientPredicateCollection = new ClientPredicateCollection();
+        PredicateCollection<Client> predicateCollection = new PredicateCollection<>();
 
-        addClientPredicate(clientPredicateCollection, argMultimap,
-                           PREFIX_CLIENT_NAME, NameContainsKeywordsPredicate::new);
-        addClientPredicate(clientPredicateCollection, argMultimap,
-                           PREFIX_CLIENT_EMAIL, EmailContainsKeywordsPredicate::new);
-        addClientPredicate(clientPredicateCollection, argMultimap,
-                           PREFIX_CLIENT_PHONE, PhoneContainsKeywordsPredicate::new);
-        addClientPredicate(clientPredicateCollection, argMultimap,
-                           PREFIX_CLIENT_ADDRESS, AddressContainsKeywordsPredicate::new);
+        addClientPredicate(predicateCollection, argMultimap, PREFIX_CLIENT_NAME, Client::getName);
+        addClientPredicate(predicateCollection, argMultimap, PREFIX_CLIENT_EMAIL, Client::getEmail);
+        addClientPredicate(predicateCollection, argMultimap, PREFIX_CLIENT_PHONE, Client::getPhone);
+        addClientPredicate(predicateCollection, argMultimap, PREFIX_CLIENT_ADDRESS, Client::getAddress);
 
-        if (clientPredicateCollection.hasNoPredicates()) {
+        if (predicateCollection.hasNoPredicates()) {
```

Comment:
Could you help add this check in for both `FindClientCommandParser.java` and `FindOrderCommandParser.java`?
```suggestion
        if (predicateCollection.hasNoPredicates() || !argMultimap.getPreamble().isEmpty()) {
```